rectools/dataset/dataset.py

```diff
@@ -15,18 +15,75 @@
 """Dataset - all data container."""
 
 import typing as tp
+from collections.abc import Hashable
 
 import attr
 import numpy as np
 import pandas as pd
+import typing_extensions as tpe
+from pydantic import PlainSerializer
 from scipy import sparse
 
 from rectools import Columns
+from rectools.utils.config import BaseConfig
 
-from .features import AbsentIdError, DenseFeatures, Features, SparseFeatures
+from .features import AbsentIdError, DenseFeatures, Features, SparseFeatureName, SparseFeatures
 from .identifiers import IdMap
 from .interactions import Interactions
 
+AnyFeatureName = tp.Union[str, SparseFeatureName]
+
+
+def _serialize_feature_name(spec: tp.Union[AnyFeatureName, tp.Any]) -> Hashable:
+    if isinstance(spec, tuple):
+        return tuple(_serialize_feature_name(item) for item in spec)
+    if isinstance(spec, (int, float, str, bool)):
+        return spec
+    if not isinstance(spec, np.ndarray) and np.issubdtype(spec, np.number) or np.issubdtype(spec, np.bool_):
+        return spec.item()
+    raise ValueError(
+        f"""
+        Serialization for feature name {spec} is not supported.
+        Please convert your feature names and values to strings, numbers, booleans or their tuples or lists.
```

Comment:
1. Why are you talking about values here?
2. I can't see you support lists

Reply:
Values are part of feature names for category features. Added this to description

Reply:
Dropped lists support from message because we expect Hashable on return